src/Functions/FunctionsJSON.h

```diff
@@ -286,27 +287,78 @@ class FunctionJSON : public IFunction, WithContext
     String getName() const override { return Name::name; }
     bool isVariadic() const override { return true; }
     size_t getNumberOfArguments() const override { return 0; }
+
+    // Both NULL and JSON NULL should generate NULL value.
+    bool useDefaultImplementationForNulls() const override { return false; }
+
     bool useDefaultImplementationForConstants() const override { return true; }
     bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
 
     DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
     {
-        return Impl<DummyJSONParser>::getReturnType(Name::name, arguments);
+        if (arguments.empty())
+            throw Exception{
+                "Function " + String(Name::name) + " cannot have empty arguments", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH};
+
+        auto return_type = Impl<DummyJSONParser>::getReturnType(Name::name, arguments);
+        if (arguments[0].type->isNullable() && !return_type->isNullable())
+            return std::make_shared<DataTypeNullable>(return_type);
+        return return_type;
     }
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
     {
+        if (arguments[0].type->onlyNull())
+            return result_type->createColumnConstWithDefaultValue(input_rows_count);
+
+        ColumnPtr nested_result;
+        ColumnPtr result_null_map_column;
+        const ColumnsWithTypeAndName * src_arguments = &arguments;
+        ColumnsWithTypeAndName nested_arguments;
+        if (const auto * nullable = checkAndGetColumn<ColumnNullable>(*arguments[0].column))
+        {
+            result_null_map_column = nullable->getNullMapColumnPtr();
+            auto args = arguments.size();
+            nested_arguments.reserve(args);
+            nested_arguments.push_back(columnGetNested(arguments[0]));
```

Comment:
Shouldn't we ensure that there's no gabage on null places and executing function on this values won't throw error?

Reply:
Hmm, I don't understand.

Reply:
I mean, we get nested column and pass in to function, but we don't know which values in nested column behind NULL values (like `nested column: ('{}', ''), nullmap: (0, 1)`). But seems that function returns null in that case anyway event if input is not valid.

Reply:
Ok. But I guess we don't need to support Nullables for arguments except the first one.